clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -9768,7 +9768,7 @@ def err_builtin_launder_invalid_arg : Error<
 // SYCL-specific diagnostics
 def err_sycl_attribute_address_space_invalid : Error<
   "address space is outside the valid range of values">;
-def err_sycl_kernel_name_class_not_top_level : Error<
+def warn_sycl_kernel_name_class_not_top_level : Warning<
```

Comment:
Why does the error gets converted to a warning? *if* the kernel is named, the error still applies...